homework/index.js

```diff
@@ -23,24 +10,151 @@
+        });
+        imgDiv.firstChild.style.maxHeight = '50px';
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
+      if (infoRight.firstElementChild === null) {
+        createAndAppend('p', infoRight, {
+          class: 'no-content',
+          text: 'There are no contributions yet.',
+        });
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+    }
+
+    fetchJSON(urlText)
+      .then(goodResult => renderContributions(goodResult))
+      .catch(somethingWrong => renderError(somethingWrong));
+  }
+
+  function showRepoInfo(wantedRepo) {
+    const infoLeft = createAndAppend('div', infoContainer, {
+      class: 'info-left',
+      text: 'Repository information:',
     });
+    const table = createAndAppend('table', infoLeft);
+    const tbody = createAndAppend('tbody', table);
+    addRow(tbody, `Name`, wantedRepo.name);
+    addRow(tbody, `Description`, wantedRepo.description);
+    addRow(tbody, `Forks`, wantedRepo.forks);
+    addRow(tbody, `Last update`, new Date(wantedRepo.updated_at).toLocaleString());
+    const toBeLink = tbody.firstChild.childNodes[1];
+    toBeLink.innerHTML = `<a href="${wantedRepo.html_url}" target="_blank">${wantedRepo.name}</a>`;
+    const descriptionCell = tbody.childNodes[1].childNodes[1];
+    if (descriptionCell.textContent === '') {
+      createAndAppend('span', descriptionCell, {
+        class: 'no-content',
+        text: 'There is NO description',
+      });
+    }
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  const clearinfoContainer = () => {
+    while (infoContainer.firstChild) {
+      infoContainer.removeChild(infoContainer.firstChild);
+    }
+  };
+
+  function starter(url) {
+    function renderResult(repositories) {
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      repositories.forEach((repo, index) => {
+        createAndAppend('option', selectEl, { text: repo.name, value: index });
+      });
+      showRepoInfo(repositories[selectEl.value]);
+      showContributions(repositories[selectEl.value].contributors_url);
+      selectEl.addEventListener('change', () => {
+        clearinfoContainer();
+        showRepoInfo(repositories[selectEl.value]);
+        setTimeout(showContributions(repositories[selectEl.value].contributors_url), 0);
+      });
+    }
+    fetchJSON(url)
+      .then(goodResult => renderResult(goodResult))
+      .catch(somethingWrong => renderError(somethingWrong));
```

Comment:
Funny, but not the best names... Describe the content of the data: `repositories`, `error`.

Reply:
I wrote so in order not te be confused with line 141. They are parameters with the same name. But now I got the logic .